client/modules/IDE/hooks/useWhatPage.js

```diff
@@ -0,0 +1,33 @@
+/* eslint-disable no-useless-escape */
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useLocation } from 'react-router-dom';
+
+/**
+ *
+ * @returns {"home" | "myStuff" | "login" | "signup" | "account" | "examples"}
+ */
+const useWhatPage = () => {
+  const projectName = useSelector((state) => state.project.name);
+  const username = useSelector((state) => state.user.username);
+  const { pathname } = useLocation();
+
+  const [pageName, setPageName] = useState(projectName);
```

Comment:
This doesn't really need to be a state.  You could use a `useMemo` instead of `useState` + `useEffect`, like we did in another place.

Reply:
yeah